pyqtorch/utils.py

```diff
@@ -231,6 +235,41 @@ def promote_operator(operator: Tensor, target: int, n_qubits: int) -> Tensor:
     return operator
 
 
+def random_dm_promotion(target: int, dm_input: DensityMatrix, n_qubits: int) -> DensityMatrix:
+    """
+    Promotes the density matrix `dm_input` at the specified target qubit with random states.
+
+    Args:
+        target (int): The index of the target qubit where the promotion will be applied.
+        dm_input (DensityMatrix): The input density matrix to promote.
+        n_qubits (int): The total number of qubits in the quantum system.
+
+    Returns:
+        DensityMatrix: The density matrix after applying random promotion.
+
+    Raises:
+        ValueError: If `target` is not within the valid range [0, n_qubits - 1].
+    """
+    if target > n_qubits - 1:
+        raise ValueError("The target must be a valid qubit index within the circuit's range.")
+    torch.manual_seed(
```

Comment:
I would prefer to leave it here because I want this function to be general. I believe it is quite useful and could be used in other cases beyond the tests where I have used it so far. That's why I wrote it in `utils.py` instead of `tests`.

Reply:
I don't think it's good practise to hide a seed set like this as that is a global function, i.e. it's not limited to the scope of the function that you set it in and will impact the rest of the program.

Reply:
I’ve revised the function to use a local random number generator as 
```python3
rng = torch.Generator().manual_seed(n_qubits)
```